### PR TITLE
Host ip lookup optimization

### PR DESCRIPTION
**Pull request by zzxgzgz**
This PR does the following:

- Added more logs in `update_neighbor_state_workitem(NeighborState current_NeighborState, GoalStateV2 &parsed_struct, GoalStateOperationReply &gsOperationReply)` to furthure identify the latency.
- Replaced `aca_is_port_on_same_host`, which caused a lot of latency(can be ~50ms) when updating a neighbor.
- Formatted some code, so that it looks better.

Performance:

After this change, for the concurrent ping test with 200 ports in the each host, the avg ping speed can be reduced to 70 ~ 80 ms.

Potential future update:
There is another latency when ACA calls `alcor::schema::NeighborConfiguration current_NeighborConfiguration = current_NeighborState.configuration();`, this latency can also take a lot of time when updating a neighbor. However, this PR does not address this problem, as time is limited.

### CODE REVIEW

File: src/ovs/aca_ovs_l2_programmer.cpp [hunk 1]
```diff
@@ -21,6 +21,16 @@
 #include <chrono>
 #include <thread>
 #include <errno.h>
+#include <boost/algorithm/string/classification.hpp> // Include boost::for is_any_of
```
Comment (xieus) [inline src/ovs/aca_ovs_l2_programmer.cpp:24]: Does this require changing make files or build script of ACA?

File: src/ovs/aca_ovs_l2_programmer.cpp [hunk 2]
```diff
@@ -84,6 +93,74 @@ ACA_OVS_L2_Programmer &ACA_OVS_L2_Programmer::get_instance()
...
+                   host_ip) != this->host_ips_vector.end();
+}
+
+void ACA_OVS_L2_Programmer::get_local_host_ips()
+{
+  struct ifaddrs *ifaddr, *ifa;
+  int family, s, n;
+  char host[NI_MAXHOST];
+
+  if (getifaddrs(&ifaddr) == -1) {
```
Comment (xieus) [inline src/ovs/aca_ovs_l2_programmer.cpp:108]: Which lib offers getifaddrs func?

File: src/ovs/aca_ovs_l2_programmer.cpp [hunk 3]
```diff
@@ -84,6 +93,74 @@ ACA_OVS_L2_Programmer &ACA_OVS_L2_Programmer::get_instance()
   return instance;
 }
 
+bool ACA_OVS_L2_Programmer::is_ip_on_the_same_host(const std::string host_ip)
+{
+  return std::find(this->host_ips_vector.begin(), this->host_ips_vector.end(),
+                   host_ip) != this->host_ips_vector.end();
+}
+
+void ACA_OVS_L2_Programmer::get_local_host_ips()
```
Comment (xieus) [inline src/ovs/aca_ovs_l2_programmer.cpp:102]: If this method comes from a third-party lib, the best practice is to acknowledge and put the license statement of third party lib in our license statement in this file.

**Review by xieus — COMMENTED** [review]
@zzxgzgz A few comments.

File: src/ovs/aca_ovs_l2_programmer.cpp (hunk 1 above)
Comment (zzxgzgz) [reply src/ovs/aca_ovs_l2_programmer.cpp:24]: I don't think so, I use the same script and it works just fine.

**Review by zzxgzgz — COMMENTED** [review]
Updated code from master, should be ready to be merged.

**Review by xieus — APPROVED** [review]
LGTM

**Pull request merged**